### PR TITLE
Add tests to increase TypeChecker.cpp coverage

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol [hunk 1]
```diff
@@ -0,0 +1,5 @@
+contract test {
+    function f() public returns (uint256 r, uint8) { return ((12, "")); }
+}
+// ----
+// TypeError: (69-75): Return argument type bool is not implicitly convertible to expected type (type of first return variable) uint256.
```
Comment (chriseth) [inline test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol]: Interesting - where does the `bool` come from?

File: test/libsolidity/syntaxTests/types/var_type_invalid_rational.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
+contract C {
+    function h() internal pure returns (uint, uint, uint) {
```
Comment (chriseth) [inline test/libsolidity/syntaxTests/types/var_type_invalid_rational.sol]: Are these functions needed?

File: test/libsolidity/syntaxTests/types/var_type_suggest.sol [hunk 3]
```diff
@@ -6,6 +6,7 @@ contract C {
         return x;
     }
     function f() internal pure {
+        var s = -31415;
```
Comment (chriseth) [inline test/libsolidity/syntaxTests/types/var_type_suggest.sol]: Why did you remove this?

File: test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol (hunk 1 above)
Comment (Marenz) [reply test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol]: hu.. good question.

File: test/libsolidity/syntaxTests/types/var_type_invalid_rational.sol (hunk 2 above)
Comment (Marenz) [reply test/libsolidity/syntaxTests/types/var_type_invalid_rational.sol]: Ah.. probably not, it's just a modified copy of the test below

File: test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol (hunk 1 above)
Comment (Marenz) [reply test/libsolidity/syntaxTests/nameAndTypeResolution/typeChecking/return_tuple_not_convertible.sol]: Was my fault. Didn't update the expectation after experimenting

**Comment by Marenz** [comment]
Updated:
* Fixed wrong expectation
* Removed unneeded functions

**Review by chriseth — APPROVED** [review]
